packages/material-ui/src/Modal/ModalManager.test.js

```diff
@@ -161,17 +161,17 @@ describe('ModalManager', () => {
       const modal2 = {};
       modalManager.add(modal1, container3);
       modalManager.mount(modal1);
-      assert.strictEqual(container3.children[0].getAttribute('aria-hidden'), 'true');
+      expect(container3.children[0]).to.be.ariaHidden;
```

Comment:
What do you think of minimizing the API surface people have to know for contributing? Where does .ariaHidden come from? Would it be simpler with the native DOM API?

```suggestion
      expect(container3.children[0].getAttribute('aria-hidden')).to.equal('true');
```

Reply:
Ok, I have read the rest of the pull request, I'm more confused. What do you think of importing the aria-hidden helper function instead?

Reply:
Doesn't scale very well (need to import all the matchers) and doesn't address the concern (knowledge is required in both cases). It's common to rely on injected matcher methods. Autocomplete should work (needs other PR) which helps with discoverability and teaching.

Reply:
Oh and yeah it helps debugging failing test. A simple boolean helper would not.

Reply:
> need to import all the matchers

I would hope a file test doesn't need to import all the matchers. For instance, will we need the aria-hidden helper outside of the modal context?

> knowledge is required in both cases

It depends on the case. When writing the test case, knowledge is required in both cases. When reading the test case, the import can be followed. Considering we spend most of our time reading the test case, I find that value proposing interesting.

> Oh and yeah it helps debugging failing test. A simple boolean helper would not.

Yeah, that wasn't a good idea from my part. However, does it invalidate the import helper point? Can we write the helper to run the assertions? I believe we can.

Reply:
> Considering we spend most of our time reading the test case, I find that value proposing interesting.

So how exactly is one less readable than the other
```diff
expect(element).to.be.ariaHidden;
assert.strictEqual(ariaHidden(element), true);
```

When you read the test you don't need to know how something is implemented. That would distract from the test. It is intended that logic is hidden so that you can think about how a component behaves not how this behavior is checked.

Reply:
I was considering the: "understand how the code works" dimension.

Reply:
> When it breaks, we often need to understand how it works.

This is the fundamental issue. When something fails you need to understand how the code works not your test. 

I understand that this was often not the case previously with all the enzyme find/instance/simulate shenanigans but we're writing actual tests now not just fancy snapshot tests.

Reply:
And tests are not always regression test. Most of the time you write code to not regress from the test behavior not tests to regress from code behavior. TDD is a prominent example for the first approach. Snapshot testing for the latter.

Reply:
There's a reason why we say "test" and "regression test". One being more specific and less common therefore having the longer word. Regression test make little sense for behavior that's not observable to the user (e.g. component snapshot test, asserting instance fields).

Reply:
I have done a quick search on the topic. It seems that people define "regression test" as "when it fails, it tells you that the application no longer behaves the way it used to".
I was using the term in a broader context, closer to "acceptance test": "when it fails, it tells you that the application is not doing what the customer expects it to do".

Reply:
exactly. And these tests next to our components are acceptance test that describe what's expected. The more declarative the better.

Reply:
The only regression test I'm aware of that is valuable is the screenshot tests. I would be cautious with snapshot tests, it can quickly end-up like this: they take too much time to look, we accept the changes blindly.